sdk/rust/oak_runtime_compat/src/asylo/allocator.rs

```diff
@@ -73,3 +76,21 @@ unsafe impl GlobalAlloc for System {
         free(ptr);
     }
 }
+
+// Define what happens in an Out Of Memory (OOM) condition.
+#[alloc_error_handler]
+fn alloc_error(_layout: Layout) -> ! {
+    unsafe { asylo::abort() }
+}
+
+// See https://doc.rust-lang.org/nomicon/panic-handler.html.
+#[panic_handler]
+fn panic(_info: &PanicInfo) -> ! {
+    unsafe { asylo::abort() }
+}
+
+/// Provide the entrypoint needed by the compiler's failure mechanisms when
+/// `std` is unavailable.  See ["No
+/// stdlib" documentation](https://doc.rust-lang.org/1.2.0/book/no-stdlib.html).
+#[lang = "eh_personality"]
+pub extern "C" fn eh_personality() {}
```

Comment:
I would expect this to be defined in `sdk/rust/oak_runtime_compat/src/lib.rs`, where `no_std` is also specified? It does not seem specific to this allocator, or is it? And similarly for the panic handler.
Not sure about `alloc_error`, if possible I would leave that out of this crate too.

Basically, simply importing a crate should have the least amount of side effects possible (ideally none).

Reply:
I've moved `eh_personality` to `lib.rs`, and `panic` and `alloc_error` to `asylo/mod.rs` (as they are implementation specific).